packages/components/src/custom-select-control-v2/styles.ts

```diff
@@ -132,3 +110,42 @@ export const SelectedItemCheck = styled( Ariakit.SelectItemCheck )`
 	margin-inline-start: ${ ITEM_PADDING };
 	font-size: 24px; // Size of checkmark icon
 `;
+
+const truncateStyles = css`
+	overflow: hidden;
+	text-overflow: ellipsis;
+	white-space: nowrap;
+`;
+
+export const ItemHintWrapper = styled.div`
+	display: flex;
+	justify-content: space-between;
+	align-items: center;
+	flex-wrap: wrap;
+	flex: 1;
+`;
+
+export const SelectedExperimentalHintWrapper = styled.div`
+	${ truncateStyles }
+`;
+
+export const SelectedExperimentalHintItem = styled.span`
+	color: ${ COLORS.theme.gray[ 600 ] };
+	margin-inline-start: ${ space( 2 ) };
+`;
+
+export const ItemHintContent = styled.span`
+	padding-inline-end: ${ space( 4 ) };
+`;
+
+export const ItemHint = styled.span`
+	// Necessary to override V1 styles passed via the legacy
+	// '.components-custom-select-control__item-hint' class
+	${ SelectItem } && {
+		color: ${ COLORS.theme.gray[ 600 ] };
+		text-align: initial;
+		line-height: ${ CONFIG.fontLineHeightBase };
+		padding-inline-end: ${ space( 1 ) };
+		margin-block: ${ space( 1 ) };
```

Comment:
Adding a bit of block margin improves the vertical spacing when the hint takes its own separate line. But the amount of margin is not enough to increase the row height when the hint is on the same line as the option item content. Applying both top and bottom margin ensures good vertical centering too.

Reply:
We're not adding `block-margin` anymore aftering using the `row-gap` and `column-gap` props (see discussion [here](https://github.com/WordPress/gutenberg/pull/62848#discussion_r1660962088))